### PR TITLE
Homework 11

### PR DESCRIPTION
**Pull request by jurisk**

### CODE REVIEW

File: homeworks/11-animations/index.css [hunk 1]
```diff
@@ -0,0 +1,98 @@
+* {
+  padding: 0;
+  margin: 0;
+  --delay: 180ms;
```
Comment (avilde) [inline homeworks/11-animations/index.css:4]: Common approach is to use `:root` selector to define scope for CSS variables.

https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#basic_usage

File: homeworks/11-animations/index.css [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  background-color: #f0f0f0;
+}
+
+#middleCircleArc {
+  animation: middleCircleArcMovement calc(var(--delay) * 5) cubic-bezier(.17,.67,.83,.67) infinite;
+  transform-origin: center center;
+}
+
+#innerCircleArc {
+  animation: innerCircleArcMovement calc(var(--delay) * 7) linear infinite;
```
Comment (avilde) [inline homeworks/11-animations/index.css:22]: Good job on re-using `delay` value 👍🏻

File: homeworks/11-animations/index.css [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  width: 100%;
+  height: 100vh;
+  justify-content: center;
+  align-items: center;
+  display: flex;
+  background-color: #f0f0f0;
+}
+
+#middleCircleArc {
+  animation: middleCircleArcMovement calc(var(--delay) * 5) cubic-bezier(.17,.67,.83,.67) infinite;
```
Comment (avilde) [inline homeworks/11-animations/index.css:17]: In CSS common code style practice is to add space after comma for readability `cubic-bezier(.17, .67, .83, .67)`

File: homeworks/11-animations/index.html [hunk 4]
```diff
@@ -4,6 +4,124 @@
...
+            a 180,180 0 0,1 180,180
+          "
+          stroke="#606060"
+          stroke-width="2"
+        />
+
+        <circle id="innerCircle" cx="300" cy="300" r="160" stroke="#c0c0c0" stroke-width="6"/>
+        <path
+          id="innerCircleArc"
+          d="
```
Comment (avilde) [inline homeworks/11-animations/index.html:115]: The path `d` attribute formatting seems random in all file.

File: homeworks/11-animations/index.html [hunk 5]
```diff
@@ -4,6 +4,124 @@
   <link rel="stylesheet" href="index.css">
 </head>
 <body>
+  <div class="container">
+    <svg width="600" height="600" fill="none">
+      <g class="hexes">
+        <!-- Hex shape taken from https://weareoutman.github.io/rounded-polygon/ -->
+        <!-- I am aware that I really should render these dynamically instead of copy-pasting, but I wanted to keep this pure HTML + CSS -->
```
Comment (avilde) [inline homeworks/11-animations/index.html:11]: SVGs are unique in a way as you can re-use certain parts of your SVG by using `x-link` to element id.

https://css-tricks.com/use-and-reuse-everything-in-svg-even-animations/

```html
      <defs>
        <path
          id="hex"
          stroke="none"
          fill="#606060"
          d="M 300 300 m -32 -32 m27.5 3.3867513459481a10 10 0 0 1 10 0l17.712812921102 10.226497308104a10 10 0 0 1 5 8.6602540378444l0 20.452994616207a10 10 0 0 1 -5 8.6602540378444l-17.712812921102 10.226497308104a10 10 0 0 1 -10 0l-17.712812921102 -10.226497308104a10 10 0 0 1 -5 -8.6602540378444l0 -20.452994616207a10 10 0 0 1 5 -8.6602540378444"
        />
      </defs>

      <g class="hexes">
        <!-- Hex shape taken from https://weareoutman.github.io/rounded-polygon/ -->
        <use xlink:href="#hex" id="center" class="hex" />
        <use xlink:href="#hex" id="west" class="hex" transform="translate(-68, 0)" />
        <use xlink:href="#hex" id="east" class="hex" transform="translate(68, 0)" />
        <use xlink:href="#hex" id="north-west" class="hex" transform="translate(-32, -60)" />
        <use xlink:href="#hex" id="north-east" class="hex" transform="translate(32, -60)" />
        <use xlink:href="#hex" id="south-west" class="hex" transform="translate(-32, 60)" />
        <use xlink:href="#hex" id="south-east" class="hex" transform="translate(32, 60)" />
      </g>
```

Keep in mind that defining transform property inline will override your  animation transform (which is scaling), to overcome this you can individually define `scale` as property.

```css
@keyframes hexAnimation {
  0% {
    opacity: 0;
    scale: 0.5;
  }
  8% {
    opacity: 1;
    scale: 1;
  }
  50% {
    opacity: 1;
    scale: 1;
  }
  58% {
    opacity: 0;
    scale: 0.5;
  }
}
```

https://danielcwilson.com/blog/2017/10/all-the-transform-ways/

**Review by avilde — COMMENTED** [review]
# Review

## Good parts
* Animation works
* Good naming for CSS classes and animations

## What can be improved
* re-use some parts of SVG

## Summary
Animation works well and CSS is nice and clean.
Good job, Juris!

File: homeworks/11-animations/index.html (hunk 5 above)
Comment (jurisk) [reply homeworks/11-animations/index.html]: I tried to implement this (see pushed code), but it doesn't really do the scaling animation.

Comment (avilde) [reply homeworks/11-animations/index.html]: I forgot to mention all of the CSS parts.
Change the selector from `.hexes path {` to `.hexes > * {`. The elements added to DOM using keyword `use` create a `shadow-root` element that is why the selector doesn't work.

`.hexes > * {` selector will target all descendants no matter what type.

Screenshot 2021-07-30 at 18 51 12

**Pull request merged**